app/src/main/java/fi/ohtu/mobilityprofile/remoteconnection/RequestHandler.java

```diff
@@ -112,20 +112,28 @@ private Message processStartLocationRequest() {
      * Returns a message with data that tells the most likely destination calculated in Mobility Profile.
      * @return Response message
      */
-    private Message processDestinationRequest() {
+  /*  private Message processDestinationRequest() {
         return createMessage(RESPOND_MOST_LIKELY_DESTINATION, mobilityProfile.getMostLikelyDestination(getStartLocation()));
     }
+*/
+    /**
+     * Returns a message with data that tells the most likely destinations calculated in Mobility Profile.
+     * @return Response message
+     */
+    private Message processDestinationRequest() {
+        return createMessage(RESPOND_MOST_LIKELY_DESTINATION, mobilityProfile.getListOfMostLikelyDestinations(getStartLocation()));
+    }
 
     /**
-     * Processes new routes by adding them in calendarTags or RouteSearches.
+     * Processes new routes by adding them in CalendarTags or Visits.
      *
      * @param message Message with data that tells which destination the user inputted
      */
     private void processUsedRoute(Message message) {
         Bundle bundle = message.getData();
         String destination = bundle.getString(SEND_USED_DESTINATION+"");
         if (mobilityProfile.isCalendarDestination()) {
```

Comment:
Tämä ei nyt ehdotuslistan kanssa toimi ihan näin: isCalendarDestination katsoo, onko DestinationLogit-luokan suggeestionSource == CALENDAR_SUGGESTION. Mutta nythän se on tuo vain silloin, jos mistään muusta lähteestä ei ole tullut ehdotusta. SuggestionSourceen jää lopulta tiedoksi se, mikä on viimeinen lähde, josta ollaan saatu määränpääehdotus.